### PR TITLE
Run kubelet in a job object in windows

### PR DESCRIPTION
**Pull request by marosset**
Signed-off-by: Mark Rossetti <marosset@microsoft.com>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

Improves upon https://github.com/kubernetes/kubernetes/pull/96051 by running kubelet inside a Windows job object.
Running kubelet inside a job object means all spawned process will be associated with the job object an inherit properties like process priority. 
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```
/sig windows
/sig node
/area kubelet

### CODE REVIEW

**Comment by marosset** [comment]
@jsturtevant @dcantah - could one of you review the Windows API usage?

File: cmd/kubelet/app/init_windows.go
Comment (dcantah) [inline cmd/kubelet/app/init_windows.go:54]: Examples? Didn't know kubelet spawned much 😮

Comment (marosset) [reply cmd/kubelet/app/init_windows.go:54]: - https://github.com/kubernetes/kubernetes/blob/c1e69551be1a72f0f8db6778f20658199d3a686d/pkg/kubelet/winstats/perfcounter_nodestats.go#L245
- https://github.com/kubernetes/kubernetes/blob/542829d30d2e48abb802e4f4499dc63dd8076c3d/pkg/credentialprovider/gcp/metadata.go#L125
- https://github.com/kubernetes/kubernetes/blob/c1e69551be1a72f0f8db6778f20658199d3a686d/pkg/volume/azuredd/azure_common_windows.go#L37
- https://github.com/kubernetes/kubernetes/blob/dab6f6a43d5a41ef3e2aa11d4a85c45a268ac6cd/pkg/volume/util/util.go#L664
are a few

Comment (marosset) [reply cmd/kubelet/app/init_windows.go:54]: Pretty much all volume mounting done is done with spawned powershell processes on Windows. This code is moving out of the kubelet to plugins that kubelet invokes too.

Comment (dcantah) [reply cmd/kubelet/app/init_windows.go:54]: Ahh, ok that makes sense.

File: cmd/kubelet/app/init_windows.go
Comment (dcantah) [inline cmd/kubelet/app/init_windows.go:57]: This is a bit of a mouth full, but honestly I don't know what's idiomatic here or what's "too long". Maybe this could just set up the job (create + setinformation to the pc) and return the handle. Then you can just add the current process to it in initForOS right after.

**Comment by dims** [comment]
this PR has been on hold for about 2 weeks. Can you please see how to unblock this by reviewing comments etc? or please close this out (Trying to declutter our open list of PRs)

**Comment by marosset** [comment]
> this PR has been on hold for about 2 weeks. Can you please see how to unblock this by reviewing comments etc? or please close this out (Trying to declutter our open list of PRs)

Yea, IIRC we were holding off until 1.24.
@dcantah @jsturtevant want to discuss this sometime early next week? I remember there being some extra flags needed for the same changes to merge into containerd (so we can create new job objects that were not parented to the current job object) but I don't remember if we needed that for the kubelet too.

**Comment by pacoxu** [comment]
/EasyCLA
/unhold 
for review in 1.24 release cycle.

**Comment by marosset** [comment]
@dcantah / @jsturtevant I'm going to add this to next week's SIG windows meeting for discussion.

**Comment by jsturtevant** [comment]
generally think this is a good idea. We can likely merge and iterate on this. 

We should discuss if we can eventually use something like this to implement features like  [qos](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#enabling-qos-and-pod-level-cgroups), [system-reserve](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#system-reserved), [kube-reserve](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#kube-reserved) and [enforcing node allocatable](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#enforcing-node-allocatable). I haven't spent enough time to figure this out but for instance is there value in having kubelet/containerd/etc in a single job object as it can be done with cgroups in linux?

File: cmd/kubelet/app/init_windows.go
Comment (marosset) [reply cmd/kubelet/app/init_windows.go]: Updated, PTAL

**Comment by DiptoChakrabarty** [comment]
✋ @marosset  I am from the v1.24 bug triage team , code freeze is on 03/30 so just wanted to remind to get this PR merged before that to target v1.24 :)

**Review by mrunalp — APPROVED** [review]

**Pull request merged**
